src/main/java/baseball/Application.java

```diff
@@ -1,7 +1,128 @@
 package baseball;
 
+import camp.nextstep.edu.missionutils.Console;
+import camp.nextstep.edu.missionutils.Randoms;
+import java.util.ArrayList;
+import java.util.List;
+
 public class Application {
     public static void main(String[] args) {
-        // TODO: 프로그램 구현
+        System.out.println("숫자 야구 게임을 시작합니다.");
+
+        List<Integer> computerNumberList = computerRanNumList();
+
+        boolean inGame = true;
+
+        while (inGame) {
+            System.out.println("숫자를 입력해주세요 : ");
+            String userInput = Console.readLine();
+            List<Integer> userNumberList = userInputNumList(userInput);
+
+            int strike = strikeCnt(computerNumberList, userNumberList);
+            int inList = inListCnt(computerNumberList, userNumberList);
+
+            int ball = inList - strike;
+
+            printScore(ball, strike);
+
+            if (strike == 3) {
+                System.out.println("3개의 숫자를 모두 맞히셨습니다! 게임 종료");
+                System.out.println("게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요.");
+                String againNumber = Console.readLine();
+                inGame = checkAgainGame(againNumber);
+                if (inGame) {
+                    computerNumberList = computerRanNumList();
+                }
+            }
+        }
+    }
+
+    // computer
+    private static List<Integer> computerRanNumList() {
+        List<Integer> computerNumberList = new ArrayList<>();
+        while (computerNumberList.size() < 3) {
+            int randNumber = Randoms.pickNumberInRange(1, 9);
+            if (!computerNumberList.contains(randNumber)) {
+                computerNumberList.add(randNumber);
+            }
+        }
+        return computerNumberList;
+    }
+
+    //user
+    private static List<Integer> userInputNumList(String userInput) {
+        List<Integer> userInputNumberList = new ArrayList<>();
+        checkLength(userInput);
+        for (String splitUserNumber : userInput.split("")) {
+            if (!splitUserNumber.isEmpty()) {
+                userInputNumberList.add(Integer.parseInt(splitUserNumber));
```

Comment:
"ㅂ13" 을 입력했을 경우 IllegalArgumentException이 발생해야 하는데 NumberFormatException이 발생해요! 
문자열에 대한 validation 처리가 필요할 것 같습니다! 
```suggestion
                int intValue;
                try {
                    intValue = Integer.parseInt(splitUserNumber)
                }
                catch (NumberFormatException e) {
                    throw new IllegalArgumentException("1 ~ 9 의 숫자를 입력해주세요.");
                }
                userInputNumberList.add(intValue);
                
                // 위 코드처럼 Integer.parseInt()를 try-catch 안에 넣어서 
                // 수행되지 못할 경우 IllegalArgumentException 을 발생시키도록 하면 좋을 것 같아요 :) 
```

Reply:
맞아요 ㅜㅜ 저도 이부분을 고민하다가 결국은 이상태로 제출했는데 .. 다른 분들 코드 참고하면서 이방법이 있구나하면서 배우고 있어요 !